src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -120,9 +118,11 @@ def resolve(self, root_reqs, check_supported_wheels):
             else:
                 if req.is_direct and req.name:
                     user_requested.add(canonicalize_name(req.name))
-                requirements.append(
-                    self.factory.make_requirement_from_install_req(req)
+                r = self.factory.make_requirement_from_install_req(
+                    req, requested_extras=(),
```

Comment:
Should this be...

```suggestion
                    req, requested_extras=req.extras,
```

I don't know, and my brain is too tired to figure it out.

Reply:
No, `requested_extras` should be a list of extras specified in the *parent*, to match against the extra in the markers. For example, if `package-a` has dependencies `package-b; extra == 'foo'`, the requirement object for `package-b` is created with

```python
factory.make_requirement_from_install_req(
    ireq_for_package_b,
    requested_extras=ireq_for_package_a.extras
)
```

We’re building the root requirements here, so the requested extras list is always empty.